sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -453,23 +453,26 @@ export function createConnectionContext(
   hostOrConnectionString = String(hostOrConnectionString);
 
   if (!isTokenCredential(credentialOrOptions)) {
-    const parsedCS = parseConnectionString<{ EntityPath?: string }>(hostOrConnectionString);
+    const parsedCS = parseEventHubConnectionString(hostOrConnectionString);
     if (
-      !(parsedCS.EntityPath || (typeof eventHubNameOrOptions === "string" && eventHubNameOrOptions))
+      !(
+        parsedCS.eventHubName ||
+        (typeof eventHubNameOrOptions === "string" && eventHubNameOrOptions)
+      )
     ) {
       throw new TypeError(
         `Either provide "eventHubName" or the "connectionString": "${hostOrConnectionString}", ` +
           `must contain "EntityPath=<your-event-hub-name>".`
       );
     }
     if (
-      parsedCS.EntityPath &&
+      parsedCS.eventHubName &&
       typeof eventHubNameOrOptions === "string" &&
       eventHubNameOrOptions &&
-      parsedCS.EntityPath !== eventHubNameOrOptions
+      parsedCS.eventHubName !== eventHubNameOrOptions
     ) {
       throw new TypeError(
-        `The entity path "${parsedCS.EntityPath}" in connectionString: "${hostOrConnectionString}" ` +
+        `The eventHubName "${parsedCS.eventHubName}" in connectionString: "${hostOrConnectionString}" ` +
```

Comment:
Should we retain the earlier message here? The connection string does not have the term "eventHubName", it has only the "EntityPath"